src/diff.rs

```diff
@@ -36,22 +37,51 @@ struct Diff<T> {
   new: T,
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 enum DiffStatus {
   Changed,
+  Upgraded,
+  Downgraded,
   Added,
   Removed,
 }
 
 impl DiffStatus {
   fn char(self) -> impl fmt::Display {
     match self {
-      Self::Added => "A".green(),
-      Self::Removed => "R".red(),
-      Self::Changed => "C".yellow(),
+      Self::Changed => "C".yellow().bold(),
+      Self::Upgraded => "U".bright_cyan().bold(),
+      Self::Downgraded => "D".magenta().bold(),
+      Self::Added => "A".green().bold(),
+      Self::Removed => "R".red().bold(),
     }
   }
 }
+impl cmp::Ord for DiffStatus {
+  fn cmp(&self, other: &Self) -> cmp::Ordering {
+    use DiffStatus::{
+      Added,
+      Changed,
+      Downgraded,
+      Removed,
+      Upgraded,
+    };
+    match (*self, *other) {
+      (Removed, Added) => cmp::Ordering::Greater,
+      (Removed | Added, _) => cmp::Ordering::Less,
+      (Changed | Upgraded | Downgraded, Removed | Added) => {
+        cmp::Ordering::Greater
```

Comment:
```suggestion
      (Removed | Added, _) => cmp::Ordering::Greater,
      (Changed | Upgraded | Downgraded, Removed | Added) => {
        cmp::Ordering::Less
```

Make Added and Removed be at the bottom. Also change the enum declaration and match order at the print site to match.